sklearn/linear_model/least_angle.py

```diff
@@ -1074,6 +1074,8 @@ def fit(self, X, y):
         """
         self.fit_path = True
         X, y = check_X_y(X, y, y_numeric=True)
+        X = as_float_array(X, copy=self.copy_X)
```

Comment:
Actually we could squash the three lines into 

``` X, y = check_X_y(X, y, y_numeric=True, dtype=FLOAT_DTYPES, copy=self.copy_X)```

Can I do that without any side effect?